solutions/swb-app/src/dataSets/dataSetPermissionParser.ts

```diff
@@ -0,0 +1,16 @@
+/*
+ *  Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *  SPDX-License-Identifier: Apache-2.0
+ */
+
+import { z } from 'zod';
+
+// eslint-disable-next-line @rushstack/typedef-var
+export const DataSetPermissionParser = z
+  .object({
+    subject: z.string(),
+    accessLevel: z.union([z.literal('read-only'), z.literal('read-write')])
```

Comment:
does `z.enum` not work for this?

Reply:
The accessLevel type is actualy the union of two string literals, not an enum, so I tried to match that exactly without having to add another parser.

Reply:
Interesting so this translates into `'read-only' | 'read-write'`

Reply:
`z.enum` seems to be the Zod way of creating a union of string literals. https://github.com/colinhacks/zod#zod-enums

Reply:
:-( why can't we just have nice enums like everyone else